orchagent/aclorch.cpp

```diff
@@ -3212,18 +3224,20 @@ sai_object_id_t AclOrch::getTableById(string table_id)
     }
 
     // Check if the table is a mirror table and a sibling mirror table is created
-    if (m_isCombinedMirrorV6Table &&
-            (table_id == m_mirrorTableId || table_id == m_mirrorV6TableId))
-    {
-        // If the table is v4, the corresponding v6 table is already created
-        if (table_id == m_mirrorTableId)
-        {
-            return getTableById(m_mirrorV6TableId);
-        }
-        // If the table is v6, the corresponding v4 table is already created
-        else
+    for (auto stage: {ACL_STAGE_INGRESS, ACL_STAGE_EGRESS}) {
+        if (m_isCombinedMirrorV6Table &&
+                (table_id == m_mirrorTableId[stage] || table_id == m_mirrorV6TableId[stage]))
         {
-            return getTableById(m_mirrorTableId);
+            // If the table is v4, the corresponding v6 table is already created
+            if (table_id == m_mirrorTableId[stage])
+            {
+                return getTableById(m_mirrorV6TableId[stage]);
```

Comment:
@daall looks this condition need to be swap.
if (table_id == m_mirrorTableId[stage]) shouldn't be return getTableById(m_mirrorTableId[stage]);

Reply:
@abdosi , it looks like the same logic as before and not changed in this PR

Reply:
@prsunny yes even previous logic didn't look correct to me.

Reply:
Now that you point it out, I'm not 100% sure this part is even necessary. If you look above, the sibling table gets assigned the same ID as the original table when it gets created anyways: https://github.com/Azure/sonic-swss/blob/7efc055b1be4830858a5fe364c72e0c823918e4e/orchagent/aclorch.cpp#L2726

So regardless of which ID we use, they should always be the same.

I agree with @prsunny that the logic is the same as before, so I'm inclined to keep it for now and then go back and refactor it in a separate PR (so if we discover some subtlety and this part *is* needed, we can easily revert it). Thoughts?

Reply:
ok, since the table id is same we are fine for now.